### PR TITLE
Fixes whiteships and other shuttle ruins not loading

### LINKED ISSUE #6225: Random Space shuttles won't spawn
## Reproduction: shuttles at the caravan space ruin won't spawn. probably because of some exploration fuckery

take any kind of shuttle, go to space, find the caravan ruin, the pirate ship and syndicate dropship are gone

### LINKED ISSUE #6633: Whiteship / Pirate ship / Syndicate fighters missing
I don't know if this was intentional by Bacon or overlooked when doing shuttlecruise but most of the old pilotable ships are gone, Whiteship is completely missing from all of the ruins/signals and the Pirate Ship and Syndicate little fighter that are on the Syndicate caravan attack signal are gone, only the non-pilotable ships that were destroyed are there, missing all the other loot

That explains why we haven't seen Whiteship in forever and why no one really docked randomly with a pirate ship/ Syndicate fighter outside arrivals, idk some people loved to explore and pilot them back to station, specially Whiteship

### PR DESCRIPTION
**Pull request by PowerfulBacon**
## About The Pull Request

Fixes post-round shuttle loading. Also allows multiple of the loaded shuttles to spawn in a single round.

Also adds fastdmm props to shuttle ports, to allow mappers using fastdmm to see important variables easier.

## Why It's Good For The Game

Shuttle templates do not load post-roundstart, causing whiteship and the ruin to break.

## Testing Photographs and Procedure

image
image

## Changelog
:cl:
fix: Fixes post round shuttle loading.
add: Multiple mapped shuttles can spawn per round.
fix: Fixes fastdmm not being able to compile the codebase.
add: Adds in some props to shutle ports to make fastdmm display important port variables such as width, height, dwidth and dheight first.
/:cl:

### CODE REVIEW

**Comment by PowerfulBacon** [comment]
Works fine in testmerge

**Comment by PowerfulBacon** [comment]
Docking works fine

**Comment by Fronsis** [comment]
Bless, thanks Bacon i'll try to see if i can find em today and check on their performances

File: code/_compile_options.dm [hunk 1]
```diff
@@ -108,7 +108,7 @@
 #define CBT
 #endif
 
-#if !defined(CBT) && !defined(SPACEMAN_DMM)
+#if !defined(CBT) && !defined(SPACEMAN_DMM) && !defined(FASTDMM)
```
Comment (Crossedfall) [inline code/_compile_options.dm]:
```suggestion
#if !defined(CBT) && !defined(SPACEMAN_DMM)
```

Pull the FASTDMM stuff, that editor is no longer supported.

**Review by Crossedfall — DISMISSED** [review]
FastDMM is no longer supported. Mappers should switch to https://github.com/SpaiR/StrongDMM

**Comment by Mails-z** [comment]
It has been so long since I saw the white ship. I suspected it was not appearing since I could not saw it in the signals

File: code/modules/shuttle/shuttle.dm [hunk 2]
```diff
@@ -15,6 +15,10 @@ GLOBAL_LIST_INIT(shuttle_turf_blacklist, typecacheof(list(
 	icon = 'icons/obj/device.dmi'
 	icon_state = "pinonfar"
 
+	FASTDMM_PROP(\
+		pinned_vars = list("name", "id", "width", "dwidth", "height", "dheight")\
+	)
+
```
Comment (PowerfulBacon) [inline code/modules/shuttle/shuttle.dm]:
```suggestion
```

File: code/modules/shuttle/shuttle.dm [hunk 3]
```diff
@@ -261,6 +268,10 @@ GLOBAL_LIST_INIT(shuttle_turf_blacklist, typecacheof(list(
 
 	area_type = SHUTTLE_DEFAULT_SHUTTLE_AREA_TYPE
 
+	FASTDMM_PROP(\
+		pinned_vars = list("name", "id", "dynamic_id", "width", "dwidth", "height", "dheight")\
+	)
+
```
Comment (PowerfulBacon) [inline code/modules/shuttle/shuttle.dm]:
```suggestion
```

**Review by Archanial — APPROVED** [review]

**Pull request merged**